codemod/output/SignedS3AWSRequest.ts

```diff
@@ -0,0 +1,264 @@
+      ? 0
+      : 1000 *
+          Math.min(
+            this.con.maxRetryBackoffSecs,
+            this.con.retryBackoffPower ** (this.attempts - 2)
+          );
+  }
+
+  error(reason) {
+    if (this.errorExceptionStatus()) {
+      return;
+    }
+
+    this.signer.error();
+    Global.l.d(this.request.step, "error:", this.fileUpload.id, reason);
+
+    if (typeof this.errorHandler(reason) !== "undefined") {
+      return;
+    }
+
+    this.fileUpload.warn("Error in ", this.request.step, reason);
+    this.fileUpload.setStatus(ERROR);
+    const self = this;
+    const backOffWait = this.backOffWait();
+    this.attempts += 1;
+
+    setTimeout(() => {
+      if (!self.errorExceptionStatus()) {
+        self.trySend();
+      }
+    }, backOffWait);
+  }
+
+  errorHandler(reason) {}
+
+  errorExceptionStatus() {
+    return false;
+  }
+
+  getPayload() {
+    return Promise.resolve(null);
+  }
+
+  success_status(xhr) {
+    return (
+      (xhr.status >= 200 && xhr.status <= 299) ||
+      (this.request.success404 && xhr.status === 404)
+    );
+  }
+
+  stringToSign() {
+    return encodeURIComponent(this.signer.stringToSign());
+  }
+
+  canonicalRequest() {
+    return this.signer.canonicalRequest();
+  }
+
+  signResponse(payload, stringToSign, signatureDateTime) {
+    const self = this;
+
+    return new Promise(resolve => {
+      if (typeof self.con.signResponseHandler === "function") {
+        return self.con
+          .signResponseHandler(payload, stringToSign, signatureDateTime)
+          .then(resolve);
+      }
+
+      resolve(payload);
+    });
+  }
+
+  sendRequestToAWS() {
+    const self = this;
+
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      self.currentXhr = xhr;
+      let url = [self.awsUrl, self.getPath(), self.request.path].join("");
+      const all_headers = {};
+
+      if (self.request.query_string) {
+        url += self.request.query_string;
+      }
+
+      extend(all_headers, self.request.not_signed_headers);
+      extend(all_headers, self.request.x_amz_headers);
+
+      xhr.onreadystatechange = () => {
+        if (xhr.readyState === 4) {
+          if (self.success_status(xhr)) {
+            if (
+              self.request.response_match &&
+              xhr.response.match(new RegExp(self.request.response_match)) ===
+                undefined
+            ) {
+              reject(
+                `AWS response does not match set pattern: ${self.request.response_match}`
+              );
+            } else {
+              resolve();
+            }
+          } else {
+            let reason = xhr.responseText ? getAwsResponse(xhr) : " ";
+            reason += `status:${xhr.status}`;
+            reject(reason);
+          }
+        }
+      };
+
+      xhr.open(self.request.method, url);
+      xhr.setRequestHeader("Authorization", self.signer.authorizationString());
+
+      for (const key in all_headers) {
```

Comment:
Migrating to `Fetch` would allow to reduce code complexity and in that case take advantage of the `Header` api.

Reply:
Yes, but that's not the purpose of this PR. The main goal of this PR was to improve the maintainability of the project by scoping the classes in their own files, and without changing too much of its structure, mainly to prevent the appearance of bugs